core/common/src/main/java/tachyon/wire/BlockInfo.java

```diff
@@ -0,0 +1,142 @@
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+/**
+ * The block descriptor.
+ */
+@NotThreadSafe
+public final class BlockInfo {
+  @JsonProperty("blockId")
+  private long mBlockId;
+  @JsonProperty("length")
+  private long mLength;
+  @JsonProperty("locations")
+  private List<BlockLocation> mLocations = Lists.newArrayList();
+
+  /**
+   * Creates a new instance of {@link BlockInfo}.
+   */
+  public BlockInfo() {}
+
+  /**
+   * Creates a new instance of {@link BlockInfo} from a thrift representation.
+   *
+   * @param blockInfo the thrift representation of a block descriptor
+   */
+  public BlockInfo(tachyon.thrift.BlockInfo blockInfo) {
+    mBlockId = blockInfo.getBlockId();
+    mLength = blockInfo.getLength();
+    mLocations = new ArrayList<BlockLocation>();
+    for (tachyon.thrift.BlockLocation location : blockInfo.getLocations()) {
+      mLocations.add(new BlockLocation(location));
+    }
+  }
+
+  /**
+   * @return the block id
+   */
+  public long getBlockId() {
+    return mBlockId;
+  }
+
+  /**
+   * @return the block length
+   */
+  public long getLength() {
+    return mLength;
+  }
+
+  /**
+   * @return the block locations
+   */
+  public List<BlockLocation> getLocations() {
+    return mLocations;
+  }
+
+  /**
+   * @param blockId the block id to use
+   * @return the block descriptor
+   */
+  public BlockInfo setBlockId(long blockId) {
+    mBlockId = blockId;
+    return this;
+  }
+
+  /**
+   * @param length the block length to use
+   * @return the block descriptor
+   */
+  public BlockInfo setLength(long length) {
+    mLength = length;
+    return this;
+  }
+
+  /**
+   * @param locations the block locations to use
+   * @return the block descriptor
+   */
+  public BlockInfo setLocations(List<BlockLocation> locations) {
+    Preconditions.checkNotNull(locations);
+    mLocations = locations;
+    return this;
+  }
+
+  /**
+   * @return thrift representation of the block descriptor
+   */
+  public tachyon.thrift.BlockInfo toThrift() {
+    List<tachyon.thrift.BlockLocation> locations = new ArrayList<tachyon.thrift.BlockLocation>();
+    for (BlockLocation location : mLocations) {
+      locations.add(location.toThrift());
+    }
+    return new tachyon.thrift.BlockInfo(mBlockId, mLength, locations);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof BlockInfo)) {
+      return false;
+    }
+    BlockInfo that = (BlockInfo) o;
+    return mBlockId == that.mBlockId && mLength == that.mLength
+        && mLocations.equals(that.mLocations);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(mBlockId, mLength, mLocations);
+  }
+
+  @Override
+  public String toString() {
+    return Objects.toStringHelper(this).add("id", mBlockId).add("length", mLength)
+        .add("locations", mLocations).toString();
```

Comment:
I like the style in this PR. And I think it could be a separate PR to change the style of other `toString` methods.